client/js/nonprofits/donate/parseFields/customAmounts/index.ts

```diff
@@ -0,0 +1,22 @@
+// License: LGPL-3.0-or-later
+import JsonStringParser from './JsonStringParser';
+const R = require('ramda');
+const { getDefaultAmounts } = require('../../custom_amounts');
+import { splitParam } from '..';
+
+export type Amount = number | CustomAmount;
+
+export interface CustomAmount {
+  amount: NonNullable<number>;
+  highlight: string;
+}
+
+export default function parseCustomFields(amountsString: string): Amount[] {
+  const defaultAmts = getDefaultAmounts().join();
+
+  if (amountsString.includes('{')) {
+    return new JsonStringParser(`[${amountsString}]`).results;
+  } else {
+    return R.compose(R.map(Number), splitParam)(amountsString || defaultAmts);
```

Comment:
Like we handled in parseFields/customField/{legacy, new}.ts I'd like the usage of ramda for the non JSON parser moved into the legacy version versus the the new version. This allows us to add tests to prove that our new version of the amounts parser matches the old one and we could switch entirely away from ramda at some point.

Reply:
Ok, that makes sense. I guess I'm wondering whether it makes sense to extract a out a legacy version at all in this case (it was just `R.compose(R.map(Number), splitParam)(amountsString || defaultAmts)` as an inline function in get-params before this PR. I could just use `splitParam(amountsString || defaultAmts).map(Number)` instead in customAmounts.

Reply:
I'd agree that its simple enough to not split out the legacy version.